### PR TITLE
Optimise play video, audio and makeRenderer functions

### PR DESCRIPTION
**Pull request by aravind-raveendran**
Description:

* Play, Stop Video, styling, size calculations for the VideoView's are moved to the `VideoRendererView`
* Created separate view models for ListView and SingleViews
* State maintained in one published property
* The responsibility for creating and persisting SourceViewRenderer(wrapper to MCIosViewRenderer) is done in the ViewModels

### CODE REVIEW

**Comment by CasperMok** [comment]
My app crashes occasionally in line 34, RendererRegistry.swift
[crash-log.txt](https://github.com/dolbyio-samples/rts-app-ios-viewer/files/11658794/crash-log.txt)

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/ViewModels/InteractivityViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//
+//  InteractivityViewModel.swift
+//
+
+import Foundation
+
+final class InteractivityViewModel: ObservableObject {
```
Comment (RameshDPrasad) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/ViewModels/InteractivityViewModel.swift:7]: suggestion, rename to `UserInteractionViewModel` as it confuses with the interactive viewer which means something different in this case.

**Review by RameshDPrasad — APPROVED** [review]

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/ViewModels/InteractivityViewModel.swift (hunk 1 above)
Comment (aravind-raveendran) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/ViewModels/InteractivityViewModel.swift]: Thanks @RameshDPrasad - that makes sense and updated

**Comment by aravind-raveendran** [comment]
> My app crashes occasionally in line 34, RendererRegistry.swift
> [crash-log.txt](https://github.com/dolbyio-samples/rts-app-ios-viewer/files/11658794/crash-log.txt)

@CasperMok - My version of Xcode have not caused a crash. I am thinking it could be concurrent read and write issue. I changed it to an `actor` to serialise the access and see and I could not see the crash so far.

**Pull request merged**